File: LABORATORIO_8/PREG01_LAB08-2024-2/Biblioteca/Entrada.cpp
//
// Created by USER on 18/06/2026.
//

#include "Entrada.h"

Entrada::Entrada() {
    picante=false;
}

Entrada:: ~Entrada() {

}

bool Entrada::is_picante() const {
    return picante;
}

void Entrada::set_picante(bool picante) {
    this->picante = picante;
}


void Entrada::lee(ifstream &arch) {
    char cad[50];
    Producto::lee(arch);
    arch.getline(cad,50,'\n');
    if (strcmp(cad,"picante")==0)
        picante=true;
}

void Entrada::imprime(ofstream &arch) {
    Producto::imprime(arch);

    if (picante) {
        arch<<setw(5)<<" "<<"Picante"<<endl;
    }
    else  arch<<setw(5)<<" "<<"Sin Picante"<<endl;

}

File: Restaurante_2024-2/src/Entrada.java
import java.util.Scanner;

public class Entrada extends Producto{
    private boolean picante;

    public boolean isPicante() {
        return picante;
    }

    public void setPicante(boolean picante) {
        this.picante = picante;
    }

    //142 B Cafe 2 grande
    //142 B Agua_Mineral 2.5 pequeno
    //142 E Quesadillas 5.5 sin_picante

    @Override
    public String tiposoy(){
        return "E";
    }

    @Override
    public void lee(Scanner arch){
        String pica;
        super.lee(arch);
        pica = arch.next();
        if(pica.compareTo("picante")==0)
            picante=true;
    }

    @Override
    public void imprime(){
        super.imprime();
        if(picante)
            System.out.println("Picante");
        else
            System.out.println("Sin picante");

    }

}
